Thunder/__main__.py

```
@@ -252,12 +283,23 @@ async def start_services():
+            logger.error(f"Error during canonical touch task cleanup: {e}", exc_info=True)
+
         if 'app_runner' in locals() and app_runner is not None:
             try:
                 await app_runner.cleanup()
             except Exception as e:
                 logger.error(f"Error during web server cleanup: {e}")
 
+        try:
+            await db.close()
+            print("   ✓ Database connection closed")
+        except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)